src/upgradeConfig.js

```diff
@@ -1,110 +1,81 @@
-const { presets: oldPresets, plugins: oldPlugins } = require('./packageData');
+const packageData = require('./packageData');
 
-// TODO: fix all of this
-function changePresets(config, options = {}) {
-  let presets = config.presets;
+function parseConfigCollection(collection) {
+  if (!Array.isArray(collection) && typeof collection === 'string') {
+    return collection.split(',').map(collection => collection.trim());
+  }
+
+  return collection;
+}
+
+function parseConfigItem(rawConfigItem, configItemType) {
+  let name = Array.isArray(rawConfigItem) ? rawConfigItem[0] : rawConfigItem;
 
-  if (!Array.isArray(presets) && typeof presets === 'string') {
-    presets = config.presets = config.presets.split(',').map((preset) => preset.trim());
+  if (name.indexOf(`babel-${configItemType}`) !== 0 && name.indexOf('@babel/') !== 0) {
```

Comment:
Can't we use `.includes` instead of `.indexOf`? Gives a clearer intent imo.

Reply:
hmm -- i agree, but i do think we _specically_ want to check the beginning of the string, and `includes` would return true if it were anywhere in it (which would probably be a typo on the user's part, but still)

Reply:
we could use `startsWith` though? may need a polyfill depending on versions of node we're trying to support?

Reply:
ah, looks like that should be supported even in node 4. let's use `startsWith` here